template/runtime/src/lib.rs

```diff
@@ -303,8 +303,27 @@ impl evm::Trait for Runtime {
 	type Precompiles = ();
 }
 
+pub struct EthereumFindAuthor;
+impl FindAuthor<H160> for EthereumFindAuthor {
+	fn find_author<'a, I>(digests: I) -> Option<H160> where
+		I: 'a + IntoIterator<Item=(ConsensusEngineId, &'a [u8])>
+	{
+		for (id, mut data) in digests.into_iter() {
+			if id == AURA_ENGINE_ID {
```

Comment:
I think it would be better if we implement this as a wrapper of the real `FindAuthor` type.

```
pub struct EthereumFindAuthor<T, F: FindAuthor<T::AccountId>>(PhantomData<(T, F)>);
```

The implementation then uses the one of `F`, and then convert that `T::AccountId` into `H160`.

Reply:
The original `FindAuthor` trait implementation for both Aura and Babe returns the autority index, not the AuthorityId / AccountId. So it looks like:

`impl FindAuthor<u32> for Module<T>{}
`

If we want to wrap the original implementation - dropping `T` bound as no access to associated types are needed - in our case would look like this (for Aura):

```
pub struct EthereumFindAuthor<F>(PhantomData<F>);
impl<F: FindAuthor<u32>> FindAuthor<H160> for EthereumFindAuthor<F>
{
	fn find_author<'a, I>(digests: I) -> Option<H160> where
		I: 'a + IntoIterator<Item=(ConsensusEngineId, &'a [u8])>
	{
		if let Some(author_index) = F::find_author(digests) {
			let authority_id = Aura::authorities()[author_index as usize].clone();
			return Some(HashTruncateConvertAccountId::<BlakeTwo256>::convert_account_id(&authority_id));
		}
		None
	}
}
```

and then:

`type FindAuthor = EthereumFindAuthor<Aura>;
`

So we still need to explicitly use the `Aura` module inside the implementation to access the `authorities` Storage, but there will be a slight improvement as there is no need to match id against `AURA_ENGINE_ID` or `BABE_ENGINE_ID`.

Reply:
Solution confirmed by @sorpaas and PR updated.